app/presenters/two_factor_auth_code/generic_delivery_mode_presenter.rb

```diff
@@ -0,0 +1,42 @@
+module TwoFactorAuthCode
+  class GenericDeliveryModePresenter
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TranslationHelper
+    include Rails.application.routes.url_helpers
+
+    def initialize(data_model)
+      data_model.each do |k, v|
+        instance_variable_set("@#{k}", v)
+        self.class.send(:attr_reader, k.to_sym)
+      end
+    end
+
+    def header
+      raise NotImplementedError
+    end
+
+    def help_text
+      raise NotImplementedError
+    end
+
+    def authentication_fallback
+      raise NotImplementedError
+    end
+
+    def fallback_options_links
+      raise NotImplementedError
+    end
+
+    def recovery_code_link
+      t('devise.two_factor_authentication.recovery_code_fallback.text', link: recovery_code_tag)
+    end
+
+    private
+
+    def recovery_code_tag
+      content_tag(:a,
```

Comment:
Looks like `url_for` comes from `ActionView::RoutingUrlFor` but the helper method is also fine